src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowServiceTest.scala

```diff
@@ -189,4 +189,45 @@ class WorkflowServiceTest extends AsyncFlatSpec with Matchers with MockitoSugar
     result.length shouldBe 2
   }
 
+  "WorkflowService.runWorkflowsJobs" should "should insert joined dag instance when jobs ids exist in workflow" in {
+    // given
+    val underTest = new WorkflowServiceImpl(workflowRepository, dagInstanceRepository, workflowValidationService)
+
+    val workflowJoined = WorkflowFixture.createWorkflowJoined().copy()
+    val workflowId = workflowJoined.id
+    val jobsIds = workflowJoined.dagDefinitionJoined.jobDefinitions.map(_.id)
+
+    when(workflowRepository.getWorkflow(eqTo(workflowId))(any[ExecutionContext])).thenReturn(Future{workflowJoined})
+    when(dagInstanceRepository.insertJoinedDagInstance(any[DagInstanceJoined])(any[ExecutionContext])).thenReturn(Future{(): Unit})
+
+    // when
+    val result: Boolean = Await.result(underTest.runWorkflowsJobs(workflowId, jobsIds), Duration(120, TimeUnit.SECONDS))
+
+    // then
+    verify(workflowRepository, times(1)).getWorkflow(any[Long])(any[ExecutionContext])
+    verify(dagInstanceRepository, times(1)).insertJoinedDagInstance(any[DagInstanceJoined])(any[ExecutionContext])
+    result shouldBe true
+  }
+
+  "WorkflowService.runWorkflowsJobs" should "should not insert joined dag instance when jobs ids does not exist in workflow" in {
+    // given
+    val underTest = new WorkflowServiceImpl(workflowRepository, dagInstanceRepository, workflowValidationService)
+
+    val workflowJoined = WorkflowFixture.createWorkflowJoined().copy()
+    val workflowId = workflowJoined.id
+    val jobsIds: Seq[Long] = Seq(9999, 8888)
+
+    when(workflowRepository.getWorkflow(eqTo(workflowId))(any[ExecutionContext])).thenReturn(Future{workflowJoined})
+    when(dagInstanceRepository.insertJoinedDagInstance(any[DagInstanceJoined])(any[ExecutionContext])).thenReturn(Future{(): Unit})
+
+    // when
+    val result: Boolean = Await.result(underTest.runWorkflowsJobs(workflowId, jobsIds), Duration(120, TimeUnit.SECONDS))
+
+    // then
+    verify(workflowRepository, times(1)).getWorkflow(any[Long])(any[ExecutionContext])
+    verify(dagInstanceRepository, times(0)).insertJoinedDagInstance(any[DagInstanceJoined])(any[ExecutionContext])
```

Comment:
There should be `never` which you can use instead of `times(0)`

Reply:
Fixed